include/boost/next_prior.hpp

```diff
@@ -140,6 +138,24 @@ struct prior_impl1< T, Distance, true >
     }
 };
 
+template< typename T, typename Distance, bool IsIterator = is_iterator< T >::value >
+struct prior_advance_impl :
+    public prior_minus_assign_impl< T, Distance >
+{
+};
+
+template< typename T, typename Distance >
+struct prior_advance_impl< T, Distance, true >
+{
+    static T call(T x, Distance n)
+    {
+        // Avoid negating n to sidestep possible integer overflow
+        std::reverse_iterator< T > rx(x);
```

Comment:
OK, thanks for the explanation. 

> std::advance has a separate template parameter for distance

FYI, `std2::advance` (in Range TS) does not have a separate template parameter.